### ISSUE TITLE
calling .save() or is_present() methods of Resource raise exception 

### ISSUE THREAD

**Issue by mromanello** [PonteIneptique/surfrdf#1]
For the full description see [issue on upstream repo](https://github.com/cosminbasca/surfrdf/issues/19).

[Here](https://travis-ci.org/mromanello/hucit_kb/builds/492778702) you can see an example of how the code fails (this refers to the Python3 branch of `hucitlib` and uses your py3-compliant version of `surf`; however, I reproduce the same error with `surf==1.2.0` and `hucitlib` master branch. With `surf==1.1.9` all works fine).

PS: the fact that the surf repo does not have releases makes it difficult for me to do some debugging with a locally modified version of 1.1.9 and 1.2.0, and compare the two behaviours.

**Comment by mromanello** [comment]
Just a brief update: I've found the problematic line and how to make it work, but I'm not 100% sure about the logic.

The problematic line is this `if/else` block:
https://github.com/PonteIneptique/surfrdf/blob/79ffca6bd71cf5b2558aa08373cf71572fa1e64a/surf/query/__init__.py#L411-L435

Take a `statement` like 
```python
[('?s', '?p', '?o'), u'(?s = <http://purl.org/hucit/kb/authors/927/cts_urn>)']
```

In the version of `surf` that still works (`1.1.9`) the type is checked using

```python
if type(statement) in [list, tuple]
```

which for the statement above returns `False.`

In version `1.2.0` the line has been changed into 
```python
isinstance(statement, (list, tuple))
```

which now returns `True` for the same statement. If I change that line to how it previously was, all works fine, but I do not understand why this difference between `isinstance` and `type`.

PS: the type of `statement` is `NamedGroup`, which is a subclass of `Group`, which is a subclass of `list`.  Perhaps what's happening is that using `isinstance` here has the side effect of capturing this inheritance between classes (which is AFAIU unintentional)?
